be/src/exec/hdfs_scanner_text.cpp

```diff
@@ -182,15 +183,10 @@ Status HdfsTextScanner::do_open(RuntimeState* runtime_state) {
     RETURN_IF_ERROR(open_random_access_file());
     RETURN_IF_ERROR(_create_or_reinit_reader());
     SCOPED_RAW_TIMER(&_stats.reader_init_ns);
+    RETURN_IF_ERROR(_build_hive_column_name_2_index());
     for (const auto slot : _scanner_params.materialize_slots) {
-        // TODO slot maybe null?
         DCHECK(slot != nullptr);
-        ConverterPtr conv = csv::get_converter(slot->type(), true);
-        RETURN_IF_ERROR(_get_hive_column_index(slot->col_name()));
-        if (conv == nullptr) {
-            return Status::InternalError(strings::Substitute("Unsupported CSV type $0", slot->type().debug_string()));
-        }
-        _converters.emplace_back(std::move(conv));
+        _converters.emplace_back(csv::get_converter(slot->type(), true));
```

Comment:
check `options.invalid_field_as_null` here to decide whether use default converter or throw error

Reply:
fixed